### ISSUE TITLE
Determing whether or not a row mapper exists for given type when implementing RowMapperFactory

### ISSUE THREAD

**Issue by apjoseph** [jdbi/jdbi#790]
When implementing `RowMapperFactory.build(final Type type, final ConfigRegistry configRegistry)` I am looking to do a test like "If there is already a dedicated mapper of any kind registered for the provided type (or is a simple type like Integer, String, Long etc.), defer to that; otherwise return mapper from this factory" 

What is the proper way to do this? Is there a convenience method for this test that I am missing somewhere?

I'm thinking this is going to be a very common question for people writing a custom `RowMapperFactory`

**Comment by qualidafial** [comment]
When e.g. `RowMappers` searches for a mapper for a particular type, the registered factories are consulted in the opposite order they were registered, which means the last registered factory that provides a mapper for that type "wins."

Thus, if the factory is already registered with `RowMappers`, there is no way to find out if another factory can provide it.

With that said, you _can_ check `BuiltInMapperFactory` to see if it can map a particular type.

If you want to avoid overriding existing mappers, you'll need to consult the `RowMappers` registry prior to registering a new mapper to ensure you don't override it.

I'm curious what is the use case behind this question?

**Comment by apjoseph** [comment]
I was working creating a pull request to make [Rosetta](https://github.com/HubSpot/Rosetta), a library which allows you to use Jackson's `ObjectMapper` to convert JDBC results directly to beans -compatible with JDBI v3. Converting was pretty straightforward, but I ran into some issues when trying to port the [`RosettaMapperFactory`](Rosetta/RosettaJdbi/src/main/java/com/hubspot/rosetta/jdbi/RosettaMapperFactory.java) which had implemented the eradicated `ResultSetMapperFactory`.

Specifically, there was a method defined as:  

```
public boolean accepts(@SuppressWarnings("rawtypes") Class type, StatementContext ctx) {
    return !(type.isPrimitive() || type.isArray() || type.isAnnotation() || BuiltInArgumentFactory.canAccept(type));
  }
```
which, if I understand correctly, allowed the factory to refuse to yield a mapper if another `ResultSetMapper` already handled it.

I believe this was desired so that Jackson's `ObjectMapper` would be used to map a result set ONLY in cases when there wasn't another mapper for the given type explicitly defined somewhere else. You wouldn't want to go through all the overhead of Jackson to process a UUID return or other simple custom type for which a user might register an explicit mapper. 

Why I wouldn't just use the built-in reflection mapper instead, you ask? I am using a Postgres/Postgis database which has several exotic datatypes (Like Geometry and JSONB) which Jackson, via the Rosetta infrastructure, handles pretty well.  Sure I could just not take advantage of these datatypes, but then why would I have picked Postgres as a DB in the first place? Most of the simple reflection based mappers one encounters in the various other open source JDBC wrapper projects (except Hibernate -but obviously using Hibernate with JDBI defeats the entire purpose of JDBI) don't really allow you to deal efficiently with these types of situations -which is annoying if you have several tables containing these datatypes. I realize JDBI has a Postgres extension -but this isn't helpful if the datatypes actually occur as properties beans. Sadly, after years of searching, I have found no easily pluggable JDBC -> Bean converter that encompasses anywhere near the breadth of common conversion scenarios that Jackson handles for JSON (I actually find this amazing given that it is 2017 and ResultSet mapping is a pretty common need in most web applications) -I would be happy to be proved wrong about this though.
